openvalidators/reward/reward.py

```diff
@@ -104,7 +105,7 @@ def apply( self, prompt: str, responses: List[ bt.DendriteCall ], name: str) ->
 
         # Fill reward tensor.
         for idx, reward in zip(successful_completions_indices, successful_rewards):
-            filled_rewards[idx] = reward
+            filled_rewards[idx] = reward * self.model_weight
```

Comment:
It may be more fitting to move the weighting into the weighted sum in forward.py:
```python
for weight_i, reward_fn_i in zip(self.reward_weights, self.reward_functions):
        reward_i = reward_fn_i.apply( prompt, responses, name ).to( self.device )
        rewards += weight_i * reward_i
```
Check this: https://github.com/opentensor/validators/compare/sequence_rewards...feature/reward-stack-weights?expand=1

Reply:
I think the difference in the approaches is where the responsibility of the model weights should be. I took the approach of having the model as responsible for its own weighted sum (weight in), whereas what you propose is the “framework” (self) to bear this responsibility and explicitly define that in the sum (weight out).

I think both approaches are valid but I like your approach, will change the code accordingly